src/proto/streams/recv.rs

```diff
@@ -121,11 +120,12 @@ impl Recv {
     pub fn open(
         &mut self,
         id: StreamId,
+        push_promise: bool,
```

Comment:
In a few cases, passing a boolean argument is pretty clear. In this case, it's not immediately obvious what `recv.open(id, false, counts)` means. This could be an enum instead, so that it's clear at the call-site.

```rust
recv.open(id, Open::Push, counts);
// and ...
recv.open(id, Open::Headers, counts);
```